pkg/adapter/nimbus-kyverno/processor/kcpbuilder.go

```diff
@@ -39,60 +41,89 @@ func BuildKcpsFrom(logger logr.Logger, cnp *v1.ClusterNimbusPolicy) []kyvernov1.
 }
 
 // buildKpFor builds a KyvernoPolicy based on intent ID supported by Kyverno Policy Engine.
-func buildKcpFor(id string, cnp *v1.ClusterNimbusPolicy) kyvernov1.ClusterPolicy {
+func buildKcpFor(id string, cnp *v1.ClusterNimbusPolicy, rule v1.Rule) kyvernov1.ClusterPolicy {
 	switch id {
 	case idpool.EscapeToHost:
-		return clusterEscapeToHost(cnp)
+		return clusterEscapeToHost(cnp, rule)
 	default:
 		return kyvernov1.ClusterPolicy{}
 	}
 }
 
-func clusterEscapeToHost(cnp *v1.ClusterNimbusPolicy) kyvernov1.ClusterPolicy {
+func clusterEscapeToHost(cnp *v1.ClusterNimbusPolicy, rule v1.Rule) kyvernov1.ClusterPolicy {
+	var psa_level api.Level = api.LevelBaseline
 
-	labelsPerNamespace := make(map[string]map[string]string) //todo: what if we want to apply policy to  multiple resources in different namespaces? 
+	if rule.Params["psa_level"] != nil {
 
-	// Function to add or update values for a key
-	addOrUpdate := func(key string, innerMap map[string]string) {
-		if val, ok := labelsPerNamespace[key]; ok {
-			// Key exists, update the inner map
-			for k, v := range innerMap {
-				val[k] = v
-			}
-			labelsPerNamespace[key] = val
-		} else {
-			// Key does not exist, add a new entry
-			labelsPerNamespace[key] = innerMap
+		switch rule.Params["psa_level"][0] {
```

Comment:
Please add validation for this rule in the security intent controller. This validation should ensure that the psa_level is one of "restricted", "baseline", or "privileged"

Reply:
the default section will automatically handle this scenario.